src/XML/ds/X509IssuerSerial.php

```diff
@@ -0,0 +1,144 @@
+    {
+        $this->setIssuerName($name);
+        $this->setSerialNumber($serial);
+    }
+
+
+    /**
+     * Collect the value of the X509IssuerName-property
+     *
+     * @return string
+     */
+    public function getIssuerName(): string
+    {
+        return $this->X509IssuerName;
+    }
+
+
+    /**
+     * Set the value of the X509IssuerName-property
+     *
+     * @param string $name
+     */
+    private function setIssuerName(string $name): void
+    {
+        Assert::notEmpty($name, 'X509IssuerName cannot be empty');
+        $this->X509IssuerName = $name;
+    }
+
+
+    /**
+     * Collect the value of the X509SerialNumber-property
+     *
+     * @return string
+     */
+    public function getSerialNumber(): string
+    {
+        return $this->X509SerialNumber;
+    }
+
+
+    /**
+     * Set the value of the X509SerialNumber-property
+     *
+     * @param string $serial
+     */
+    private function setSerialNumber(string $serial): void
+    {
+        Assert::notEmpty($serial, 'X509SerialNumber cannot be empty');
+        $this->X509SerialNumber = $serial;
+    }
+
+
+    /**
+     * Convert XML into a X509IssuerSerial
+     *
+     * @param \DOMElement $xml The XML element we should load
+     * @return self
+     *
+     * @throws \SimpleSAML\XML\Exception\InvalidDOMElementException
+     *   If the qualified name of the supplied element is wrong
+     */
+    public static function fromXML(DOMElement $xml): object
+    {
+        Assert::same($xml->localName, 'X509IssuerSerial', InvalidDOMElementException::class);
+        Assert::same($xml->namespaceURI, X509IssuerSerial::NS, InvalidDOMElementException::class);
+
+        $issuer = XMLUtils::extractStrings($xml, AbstractDsElement::NS, 'X509IssuerName');
+        $serial = XMLUtils::extractStrings($xml, AbstractDsElement::NS, 'X509SerialNumber');
+
+        Assert::minCount($issuer, 1, MissingElementException::class);
+        Assert::minCount($issuer, 1, TooManyElementsException::class);
+
+        Assert::minCount($serial, 1, MissingElementException::class);
+        Assert::minCount($serial, 1, TooManyElementsException::class);
+
+        return new self(
+            array_pop($issuer),
+            array_pop($serial)
+        );
+    }
+
+
+    /**
+     * Convert this X509IssuerSerial element to XML.
+     *
+     * @param \DOMElement|null $parent The element we should append this X509IssuerSerial element to.
+     * @return \DOMElement
+     */
+    public function toXML(DOMElement $parent = null): DOMElement
+    {
+        $e = $this->instantiateParentElement($parent);
+
+        XMLUtils::addString($e, AbstractDsElement::NS, 'X509IssuerName', $this->X509IssuerName);
+        XMLUtils::addString($e, AbstractDsElement::NS, 'X509SerialNumber', $this->X509SerialNumber);
+
+        // Fix the xs:type on the SerialNumber
+        $e->childNodes[1]->setAttributeNS(Constants::NS_XSI, 'xsi:type', 'xs:integer');
```

Comment:
This looks fragile. I assume `XMLUtils::addString()` does not return the element itself, right?

Reply:
It actually does.. I'll use that!

Reply:
Nope, I'm open for suggestions

Reply:
Well, the most easy solution would be to just add an XML-class for this of course!